wgsl/index.bs

```diff
@@ -7913,56 +7913,64 @@ variables=] for each root identifier.
 
 Two root identifiers <dfn noexport>alias</dfn> when they have the same
 [=originating variable=].
-It is a [=dynamic error=] if:
-* more than one aliased root identifiers access the same [=memory locations=], and
-* at least one access is a [=write access|write=], and
-* both accesses occur during the execution of the same call of the function
-
-Note: This aliasing restriction applies to memory locations written by
-[=function calls=] made in the function.
-
-Note: [=Originating variables=] cannot have aliased [=memory locations=].
-See [[#var-decls]] and [[#resource-interface]].
-
-<div class='example wgsl' heading='Aliased memory views'>
+A WGSL program [=shader-creation error|must not=] contain any potential aliases.
+The program is analyzed from the leaves of the callgraph upwards (e.g. topological order).
+For each function the analysis records the following sets:
+* [=module scope|Module-scope=] variables that are [=write access|written=].
+    Including any module-scope variables that are written in functions called from this function.
+* [=module scope|Module-scope=] variables that are [=read access|read=].
+    Including any module-scope variables that are read in functions called from this function.
+* Pointer parameters that are [=write access|written=].
+* Pointer parameters that are [=read access|read=].
+
+At each [=call site=] of a function, it is a [=shader-creation error=] if either:
+* Two arguments of pointer type have the same root identifier and either
+    corresponding parameter is in the written parameter set.
+* An argument of pointer type has a root identifier that is a global variable where:
+    * the corresponding parameter is in the set of written pointer parameters
+        and the modue-scope variable is in either the written or read set of
+        module-scope variables for the called function, or
+    * the corresponding parameter is in the set of read pointer parameters and
+        the module-scope variable is in the written set of module-scope
+        variables for the called function.
+
+<div class='example wgsl' heading='Alias analysis'>
   <xmp highlight='rust'>
     var x : i32 = 0;
 
-    fn foo() {
-      bar(&x, &x); // Both p and q parameters are aliases of x.
+    fn f1(p1 : ptr<function, i32>, p2 : ptr<function, i32>) {
+      *p1 = *p2;
     }
 
```

Comment:
Suggest adding a function that propagates an aliasing hazard via its helpers:

```
fn fmiddle(p1:ptr<function,i32>, p2:ptr<function,i32>) {
   f1(p1,p2);
}
```

Reply:
The point is that an intermediate function can't sanitize or launder the behaviour of its callees